dask_sql/physical/rex/core/call.py

```diff
@@ -220,9 +220,6 @@ def __init__(self):
         super().__init__(self.cast)
 
     def cast(self, operand, rex=None) -> SeriesOrScalar:
-        if not is_frame(operand):  # pragma: no cover
-            return operand
```

Comment:
Thinking about this more, I'm wondering if it would make sense to have a separate casting function specifically for scalars? It seems like the main reason we need to explicitly cast timedeltas to `np.int64` is because they cannot be cast to pandas dtypes, but going off the `_SQL_TO_PYTHON_SCALARS` mapping we shouldn't run into this issue:

https://github.com/dask-contrib/dask-sql/blob/031c04c6e8163f54ad26b40c84e205dcec82cec7/dask_sql/mappings.py#L54-L57

Happy to figure this out in a separate PR since at the moment this seems like a one-off case, but interested to get your thoughts

Reply:
While debugging this the main issue I did run into was handling scalars since cast didn't handle scalars at all and then we implicitly end up casting the result at the end when matching the resultant types. 
I do think that a followup pr to handle scalars and see how things work when one operand is a scalar vs one being a frame can be handled. 

I'll open up an issue for tracking

Reply:
Opened #513 for tracking